modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java

```diff
@@ -1101,24 +1101,55 @@ public static Schema unaliasSchema(OpenAPI openAPI,
                         return schema; // generate a model extending map
                     } else {
                         // treat it as a typical map
-                        return unaliasSchema(openAPI, allSchemas.get(ModelUtils.getSimpleRef(schema.get$ref())),
+                        return unaliasSchema(openAPI, allSchemas.get(getSimpleRef(schema.get$ref())),
                                 importMappings);
                     }
                 }
             } else if (isObjectSchema(ref)) { // model
                 if (ref.getProperties() != null && !ref.getProperties().isEmpty()) { // has at least one property
                     return schema;
-                } else { // free form object (type: object)
-                    return unaliasSchema(openAPI, allSchemas.get(ModelUtils.getSimpleRef(schema.get$ref())),
+                } else {
+                    // free form object (type: object)
+                    if (hasValidation(ref)) {
+                        return schema;
+                    } else if (hasDescendants(simpleRef, openAPI)) {
```

Comment:
How about replacing this with:
`} else if (getAllOfDescendants(simpleRef, openAPI).size() > 0) {`
like we do in python-experimental?
That way you can delete the new hasDescendants function and use our existing getAllOfDescendants.

Reply:
I prefer the easier to read `hasDescendants()`.

`getAllOfDescendants(…).size() > 0` could easily be read as getting all descendants rather than getting the descendants for allOf composition.

Reply:
Then how about having hasDescendants call getAllOfDescendants under the hood? The code is very similar to what we already have.

Reply:
I've created `hasDescendants` because `getAllOfDescendants` is a method of DefaultCodegen that call other methods of the DefaultCodegen like `toModelName` and I only need to know if it's used once.

But maybe I can move everything in DefaultCodegen.

Reply:
This is fine as is. As you mention we only need to test presence, not qty.